lib/builder/items.rb

```diff
@@ -0,0 +1,37 @@
+module ItemBuilder
+  class Item
+    attr_accessor :items,
+                  :id,
+                  :name,
+                  :price,
+                  :quantity
+
+    def build
+      copy_from_individual_fields unless item_details_is_used?
+      items.map! do |part|
+        item = ::Item.new
+        item.id = part[:id]
+        item.name = part[:name]
+        item.price = part[:price]
+        item.quantity = part[:quantity]
+        item
+      end
+      items
+    end
+
+    private
+
+    def item_details_is_used?
+      !items.nil?
```

Comment:
maybe rename to items_exists? (or something else just don't think it's that clear at the moment)

Reply:
yes I agree, my only reasoning was following on from builder/customer.rb